heft-plugins/heft-lint-plugin/src/LinterBase.ts

```diff
@@ -156,14 +131,15 @@ export abstract class LinterBase<TLintResult> {
         cachedVersion !== version ||
         options.changedFiles.has(sourceFile)
       ) {
-        await this._measurePerformanceAsync(this._linterName, async () => {
-          const failures: TLintResult[] = await this.lintFileAsync(sourceFile);
-          if (failures.length === 0) {
-            newNoFailureFileVersions.set(relative, version);
-          } else {
-            lintFailures.push(...failures);
+        fileCount++;
+        const failures: TLintResult[] = await this.lintFileAsync(sourceFile);
+        if (failures.length === 0) {
+          newNoFailureFileVersions.set(relative, version);
+        } else {
+          for (const failure of failures) {
+            lintFailures.push(failure);
           }
-        });
+        }
       } else {
         newNoFailureFileVersions.set(relative, version);
```

Comment:
There have been a few issues about the fact that lint rules are not technically isolated, so this cache reports false negatives in a number of cases.